src/gui/EffectView.cpp

```diff
@@ -45,65 +49,60 @@
 namespace lmms::gui
 {
 
-EffectView::EffectView( Effect * _model, QWidget * _parent ) :
-	PluginView( _model, _parent ),
-	m_bg( embed::getIconPixmap( "effect_plugin" ) ),
-	m_subWindow( nullptr ),
+EffectView::EffectView(Effect * _model, QWidget * _parent) :
+	PluginView(_model, _parent),
+	m_subWindow(nullptr),
 	m_controlView(nullptr),
 	m_dragging(false)
 {
-	setFixedSize(EffectView::DEFAULT_WIDTH, EffectView::DEFAULT_HEIGHT);
 	setFocusPolicy(Qt::StrongFocus);
+	setViewWidth(EffectView::DEFAULT_WIDTH);
 
-	// Disable effects that are of type "DummyEffect"
+	m_mainLayout = new QHBoxLayout();
+	m_mainLayout->setContentsMargins(8, 2, 8, 2);
+
+	bool hasControls = effect()->controls()->controlCount() > 0;
 	bool isEnabled = !dynamic_cast<DummyEffect *>( effect() );
-	m_bypass = new LedCheckBox( this, "", isEnabled ? LedCheckBox::LedColor::Green : LedCheckBox::LedColor::Red );
-	m_bypass->move( 3, 3 );
-	m_bypass->setEnabled( isEnabled );
 
```

Comment:
Remove this line too

Reply:
Which line? The `bool isEnabled` one? That is still used in several places.

Reply:
Nope, the blank like just below.

Reply:
Oh -- I don't love removing that line since it makes the `isEnabled` seem "attached" to the code for the bypass button, when in reality it's used by the bypass button, the wet/dry, and the autoquit knob. It's not part of the configuration for the bypass button, it's part of the configuration for the whole effect just like the `hasControls` line above it.

Reply:
Okk then.